labelbox/schema/annotation_import.py

```diff
@@ -195,104 +126,188 @@ def refresh(self) -> None:
+            %s(data: {
+                %s: $parentId
+                name: $name
+                fileUrl: $fileUrl
+            }) {%s}
+        }""" % (cls.__class__.__name__, cls._mutation, cls._parent_id_field,
+                query.results_query_part(cls))
 
-    def get_parent_id(*args, **kwargs):
-        raise NotImplementedError("")
+    @classmethod
+    def _get_file_mutation(cls) -> str:
+        return """mutation create%sPyApi($parentId : ID!, $name: String!, $file: Upload!, $contentLength: Int!) {
+            %s(data: { %s : $parentId name: $name filePayload: { file: $file, contentLength: $contentLength}
+        }) {%s}
+        }""" % (cls.__class__.__name__, cls._mutation, cls._parent_id_field,
+                query.results_query_part(cls))
 
 
 class MEAPredictionImport(AnnotationImport):
-    id_name = "modelRunId"
-    import_type = ImportType.MODEL_ERROR_ANALYSIS
     model_run_id = Field.String("model_run_id")
+    _mutation = "createModelErrorAnalysisPredictionImport"
+    _parent_id_field = "modelRunId"
 
-    def get_parent_id(self):
+    @property
+    def parent_id(self) -> str:
         return self.model_run_id
 
     @classmethod
-    def create_from_file(cls, client, model_run_id, name, path):
+    def create_from_file(cls, client: "labelbox.Client", model_run_id: str,
+                         name: str, path: str) -> "MEAPredictionImport":
         return cls._create_from_file(client=client,
                                      parent_id=model_run_id,
                                      name=name,
                                      path=path)
 
     @classmethod
-    def create_from_objects(cls, client, model_run_id, name, predictions):
+    def create_from_objects(cls, client: "labelbox.Client", model_run_id: str,
+                            name, predictions) -> "MEAPredictionImport":
         return cls._create_from_objects(client, model_run_id, name, predictions)
 
+    @classmethod
+    def create_from_url(cls, client: "labelbox.Client", model_run_id: str,
+                        name: str, url: str) -> "MEAPredictionImport":
+        return cls._create_from_url(client=client,
+                                    parent_id=model_run_id,
+                                    name=name,
+                                    url=url)
+                                    
+    @classmethod
+    def from_name(
+            cls, client: "labelbox.Client", model_run_id: str,
+            name: str) -> "MEAPredictionImport":
+
+        query_str = """query getModelErrorAnalysisPredictionImportPyApi($modelRunId : ID!, $name: String!) {
+            modelErrorAnalysisPredictionImport(
+                where: {modelRunId: $modelRunId, name: $name}){
+                    %s
+                }}""" % query.results_query_part(cls)
+        params = {
+            "modelRunId": model_run_id,
+            "name": name,
+        }
+        response = client.execute(query_str, params)
+        if response is None: 
```

Comment:
This query if very similar to the MAL one. I tried to generalize the queries in the base class with `_get_file_mutation()` and `_get_url_mutation()`. Maybe you can move this to the base class since so much of it is shared with `MALPredictionImport.get_name` and just pass in the extra info? I don't feel strongly about that decision because I can imagine in the future as these become different it might be annoying. So I'll leave that up to you.

Reply:
> The tests are failing because the code needs to be formatted:
> 
> ```
> yapf tests -i --verbose --recursive --parallel --style "google"
> yapf labelbox -i --verbose --recursive --parallel --style "google"
> ```
> 
> At some point I'll add a commit hook to make this easier.

Would you like me to add this to our readme for now, then?

Reply:
lets keep them separate. If we add label imports and its a fairly different endpoint, then wll just have to pull them all out again. I think, though it increases code copying, these entities should be as separate as possible.